packages/dropdowns/src/Select/Select.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Reference } from 'react-popper';
+import { StyledInput, StyledSelect } from '../styled';
+import useDropdownContext from '../utils/useDropdownContext';
+import useFieldContext from '../utils/useFieldContext';
+
+const VALIDATION = {
+  SUCCESS: 'success',
+  WARNING: 'warning',
+  ERROR: 'error'
+};
+
+/**
+ * Applies state and a11y attributes to its children. Must be nested within a `<Field>` component.
+ */
+const Select = ({ children, ...props }) => {
+  const {
+    popperReferenceElementRef,
+    downshift: { getToggleButtonProps, getInputProps, isOpen }
+  } = useDropdownContext();
+  const { isLabelHovered } = useFieldContext();
+  const hiddenInputRef = useRef(null);
+  const triggerRef = useRef(null);
+  const previousIsOpenRef = useRef(undefined);
+
+  useEffect(() => {
+    // Focus internal input when Menu is opened
+    if (isOpen && !previousIsOpenRef.current) {
+      hiddenInputRef.current && hiddenInputRef.current.focus();
+    }
+
+    // Focus trigger when Menu is closed
+    if (!isOpen && previousIsOpenRef.current) {
+      triggerRef.current && triggerRef.current.focus();
+    }
+    previousIsOpenRef.current = isOpen;
+  }, [isOpen]);
+
+  const { ...selectProps } = getToggleButtonProps({
```

Comment:
Does this get function already return a new object? Maybe it's safe to just `const selectProps = getToggleButtonProps(...)`?